### ISSUE TITLE
Add SHA property to MergePullRequest

### ISSUE THREAD

**Issue by alfhenrik** [octokit/octokit.net#804]
The [API](https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button) caters for specifying the SHA that pull request head must match to allow merge, but the [MergePullRequest model](https://github.com/octokit/octokit.net/blob/master/Octokit/Models/Request/MergePullRequest.cs) does not, we should add this so that it can be passed in when trying to merge a PR.

**Comment by shiftkey** [comment]
:thumbsup:

**Comment by shiftkey** [comment]
Also, we should confirm that [message](https://github.com/octokit/octokit.net/blob/master/Octokit/Models/Request/MergePullRequest.cs#L22) is actually optional while we're in there...

**Comment by alfhenrik** [comment]
Yea, I don't think it is...

```
        public MergePullRequest(string message)
        {
            Ensure.ArgumentNotNull(message, "message");

            Message = message;
        }
```

**Comment by alfhenrik** [comment]
Since I want this, I'll send a PR for it

**Comment by shiftkey** [comment]
@alfhenrik it's now available in v0.12 on NuGet.

**Comment by alfhenrik** [comment]
Champion! Thanking you very much :cake:
________________________________
